RiotSwiftUI/Modules/Authentication/VerifyEmail/AuthenticationVerifyEmailViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+@available(iOS 14, *)
+typealias AuthenticationVerifyEmailViewModelType = StateStoreViewModel<AuthenticationVerifyEmailViewState,
+                                                                  Never,
+                                                                  AuthenticationVerifyEmailViewAction>
+@available(iOS 14, *)
+class AuthenticationVerifyEmailViewModel: AuthenticationVerifyEmailViewModelType, AuthenticationVerifyEmailViewModelProtocol {
+
+    // MARK: - Properties
+
+    // MARK: Private
+
+    // MARK: Public
+
+    @MainActor var completion: ((AuthenticationVerifyEmailViewModelResult) -> Void)?
+
+    // MARK: - Setup
+
+    init(emailAddress: String = "") {
+        let viewState = AuthenticationVerifyEmailViewState(bindings: AuthenticationVerifyEmailBindings(emailAddress: emailAddress))
+        super.init(initialViewState: viewState)
+    }
+
+    // MARK: - Public
+    
+    override func process(viewAction: AuthenticationVerifyEmailViewAction) {
+        switch viewAction {
+        case .send:
+            Task { await completion?(.send(state.bindings.emailAddress)) }
```

Comment:
Whilst I very much like the idea of this, I'm not sure it's worth changing for this screen as it will
- break consistency with all other SwiftUI screens which will could make maintaining it harder
- remove an easy place to insert a view action that doesn't need to interact with coordinator if one is needed in the future.

Reply:
We would still retain `Action` (some of which don't need to interact with coordinators), only cut out the `Result` type which only calls some internal function on the coordinator. So in essence the body of that internal method would be assigned directly as `actionCallback`. But as you say, it is fine to keep for consistency sake and see how it evolved with scale